### PR TITLE
Added Integration Test for upgrading subscription with a different entitlement

### PR DESCRIPTION
**Pull request by NachoSoto**
See [CF-788].
Fixed by https://github.com/RevenueCat/khepri/pull/4165.

Blocked by [TRIAGE-124].

[CF-788]: https://revenuecats.atlassian.net/browse/CF-788?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[TRIAGE-124]: https://revenuecats.atlassian.net/browse/TRIAGE-124?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by NachoSoto** [comment]
cc @joshdholtz

File: Tests/BackendIntegrationTests/StoreKitIntegrationTests.swift [hunk 1]
```diff
@@ -375,11 +375,42 @@ class StoreKit1IntegrationTests: BaseBackendIntegrationTests {
...
+            await Purchases.shared.products([
+                "com.revenuecat.monthly_9.99.super_premium.1_week_intro"
+            ]).first
+        )
+
+        let entitlements = try await Purchases.shared.purchase(product: superPremiumProduct)
+            .customerInfo
+            .entitlements
+
+        expect(entitlements.active).to(haveCount(1))
```
Comment (NachoSoto) [inline Tests/BackendIntegrationTests/StoreKitIntegrationTests.swift:397]: This fails with 2

File: Tests/BackendIntegrationTests/StoreKitIntegrationTests.swift [hunk 2]
```diff
@@ -375,11 +375,42 @@ class StoreKit1IntegrationTests: BaseBackendIntegrationTests {
...
+        expect(entitlements[Self.entitlementIdentifier]?.isActive).to(
```
Comment (NachoSoto) [inline Tests/BackendIntegrationTests/StoreKitIntegrationTests.swift:398]: This is still active.

**Comment by NachoSoto** [comment]
Looks like this is fixed: https://github.com/RevenueCat/khepri/pull/4165
Just waiting for it to be deployed to make sure this test passes now.

**Review by aboedo — APPROVED** [review]
Looks good to me, but I'd wait for @joshdholtz to also take a look before merging

**Comment by NachoSoto** [comment]
Supposedly the fix was deployed but this test is still failing :/ @joshdholtz any ideas?

**Comment by joshdholtz** [comment]
@NachoSoto Yeeaaaahhh, I re-ran it this morning and I saw that. Do these tests use the StoreKit Config file type of environment? Or are these like _actual_ _actual_ sandbox tests?

**Comment by NachoSoto** [comment]
They do use a `.storekit` file yeah.

**Comment by NachoSoto** [comment]
> My tests on simulator and actual device didn't work but its having issues validating the receipt so I need to figure out what's going on there first 

Are you using the correct API key?

**Comment by NachoSoto** [comment]
I think this is the same as #1718.

**Comment by NachoSoto** [comment]
Filed [TRIAGE-124] to look into this.

**Comment by NachoSoto** [comment]
Just rebased this to see if #1965 could have fixed this, but alas no.

**Pull request closed without merging**